docs/developer/payments/lifecycle.mdx

```diff
@@ -173,27 +258,126 @@ it may still remain unpaid if the pending amounts fail to process.
+
+Such events don't have initially any `pspReference`, it's filled in once Saleor receives a response from payment app.
+
+Saleor also creates automatically `_FAILURE` events when an exception happens:
+- when handling `transactionEventReport` mutation
+- when handling incorrect response from [sync webhook sent to payment app](/developer/extending/webhooks/synchronous-events/transaction.mdx)
+
+Such events don't have `pspReference` and include a `message` that explains the error.
+
+#### App events
+
+Events are also created by a payment app that processes the payment:
+
+- When app sends response to any of the [Transaction synchronous events](/developer/extending/webhooks/synchronous-events/transaction.mdx)
+  - Saleor will either fill missing `pspReference` on the automatically created event that already exists
+  - or will create a new event if it has different `type` (e.g. user requested `AUTHORIZATION`, but app returned `CHARGE_SUCCESS` - we'll have two events `AUTHORIZATION_REQUEST` with no `pspReference` and `CHARGE_SUCCESS`)
+- When [`transactionEventReport` mutation is called](/developer/payments/transactions.mdx#reporting-actions-for-transactions) (usually happens when payment provider sends asynchronous webhook)
+
+### Rules of calculating amounts
+
+Each event impacts different amount on `TransactionItem`:
+
+:::caution
+Each event can be excluded from calculation if it has matching `_FAILURE` event with the same `pspReference` and newer `time`.
+:::
+
+- `authorizedAmount` - `AUTHORIZATION_SUCCESS` event amount
+  - there can be only one `AUTHORIZATION_SUCCESS` event for given TransactionItem
+  - it can be modified by `AUTHORIZATION_ADJUSTMENT` event
+  - subtracting `canceledAmount`
+- `authorizePendingAmount` - sum of `AUTHORIZATION_REQUEST` event amounts
+- `chargedAmount` - sum of `CHARGE_SUCCESS` event amounts
+  - and sum `REFUND_REVERSE` event amounts
+  - subtracting all `CHARGE_BACK` events that lower this amount
+  - subtracting `refundedAmount`
+- `chargePendingAmount` - sum of `CHARGE_REQUEST` event amounts
+- `refundedAmount` - sum of `REFUND_SUCCESS` event amounts
+- `refundPendingAmount` - sum of `REFUND_REQUEST` event amounts
+- `canceledAmount` - sum of `CANCEL_SUCCESS` event amounts
+- `cancelPendingAmount` - sum of `CANCEL_REQUEST` event amounts
+
+
+:::note
+`AUTHORIZATION_ACTION_REQUIRED`, `CHARGE_ACTION_REQUIRED`, and `INFO` event types are not included in amount calculations.
+:::
+
+To learn more about each event type, see [Transaction event types](#transaction-event-types) section.
+
+#### Events deduplication
+
+Each event is identified by `pspReference`. This field is used to de-duplicate events in case there were reported multiple times by the payment app.
+
+For example if Saleor receives two `CHARGE_SUCCESS` events with the same `pspReference` and `amount` there will be only one event stored on Transaction. Second event will be treated as "already reported" and it won't be stored by Saleor.
```

Comment:
It is worth mentioning, that providing two events with CHARGE_SUCCESS, same `pspReference` but with different `amount` will cause an error.